ssl.pl

```diff
@@ -312,7 +312,30 @@
 
 ssl_context(Role, SSL, Module:Options) :-
     select_option(ssl_method(Method), Options, O1, sslv23),
-    '_ssl_context'(Role, SSL, Module:O1, Method).
+    upgrade_legacy_options(O1, O2),
+    % If the options list does not contain a cacerts/1 term (including cacerts([]))
+    % then assume that the system root certificates should be used
+    (  memberchk(cacerts(_), O2)
+    -> O3 = O2
+    ;  O3 = [cacerts([system(root_certificates)])|O2]
+    ),
+    '_ssl_context'(Role, SSL, Module:O3, Method).
+
+upgrade_legacy_options(O1, O4):-
+    % If the options list contains cacert_file/1 then update the cacerts list
+    select(cacert_file(CACertFile), O1, O2),
```

Comment:
And this select_option/3, no?

Reply:
Ok. There are some other things to take care of as well. Thanks for the feedback!

Reply:
Is there a standard way to print such a warning? A grep of lines containing both 'warning' and 'deprecated' on the source doesn't return much

Reply:
Using `print_message(warning, deprecated(Term))`.    The '$messages':deprecated//1 should become hookable as well (as prolog:deprecated//1).